tools/update-blog-dates.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * @fileoverview
+ * Updates blog-dates.json with last modified dates for blog posts.
+ * Processes staged blog files by default. Run with --init to process all files
+ * and initialize the dates file.
+ */
+
+"use strict";
+
+const fs = require("node:fs/promises");
+const path = require("node:path");
+const { execSync } = require("node:child_process");
+
+const BLOG_DIR = path.resolve(__dirname, "../src/content/blog");
+const DATES_FILE = path.resolve(__dirname, "../src/_data/blog-dates.json");
+
+/**
+ * Checks if a file exists at the given path.
+ * @param {string} filePath The path to check for file existence.
+ * @returns {Promise<boolean>} True if the file exists, false otherwise.
+ */
+async function fileExists(filePath) {
+	try {
+		await fs.access(filePath);
+		return true;
+	} catch {
+		return false;
+	}
+}
+
+/**
+ * Retrieves all blog post files from the blog directory.
+ * @returns {Promise<string[]>} Array of absolute paths to all .md files in the blog directory.
+ */
+async function getAllBlogFiles() {
+	const files = await fs.readdir(BLOG_DIR);
+	return files
+		.filter(file => file.endsWith(".md"))
+		.map(file => path.resolve(BLOG_DIR, file));
+}
+
+/**
+ * Gets the list of staged blog post files from git.
+ * @returns {Promise<string[]>} Array of absolute paths to staged .md files in the blog directory.
+ */
+async function getStagedBlogFiles() {
```

Comment:
Because we're running this with lint-staged, the staged files will be passed on the command line. You can get them using `process.argv.slice(2)`.